packages/navigator/src/browser/navigator-tree.ts

```diff
@@ -30,7 +32,102 @@ export class FileNavigatorTree extends FileTree {
     }
 
     async resolveChildren(parent: CompositeTreeNode): Promise<TreeNode[]> {
+        if (FileStatNode.is(parent)) {
+            let resolvedChildren: TreeNode[] = [];
+            let fileStat;
+            if (this.isVirtualRoot(parent)) {
+                fileStat = await this.resolveVirtualRootFileStat(parent);
+                if (fileStat) {
+                    parent.fileStat = fileStat;
+                }
+            } else {
+                fileStat = await this.resolveFileStat(parent);
+            }
+
+            if (fileStat) {
+                resolvedChildren = await this.toNodes(fileStat, parent);
+            }
+            return this.filter.filter(resolvedChildren);
+        }
         return this.filter.filter(super.resolveChildren(parent));
     }
 
+    protected async toNode(fileStat: FileStat, parent: CompositeTreeNode): Promise<FileNode | DirNode> {
```

Comment:
let's extract a new template method from `FileTree.toNode` to create an id `toNodeId(FileStat, CompositeTreeNode)`

This method should be overridden here instead of `toNode`.